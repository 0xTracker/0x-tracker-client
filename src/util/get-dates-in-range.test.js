import ms from 'ms';

import getDatesInRange from './get-dates-in-range';

it('should get dates when interval is one day', () => {
  const dateFrom = new Date('2018-01-10T00:00:00Z');
  const dateTo = new Date('2018-02-15T00:00:00Z');
  const dates = getDatesInRange(dateFrom, dateTo, ms('1 day'));

  expect(dates).toMatchSnapshot();
});

it('should get dates when interval is thirty minutes', () => {
  const dateFrom = new Date('2018-01-10T00:00:00Z');
  const dateTo = new Date('2018-01-15T00:00:00Z');
  const dates = getDatesInRange(dateFrom, dateTo, ms('30 minutes'));

  expect(dates).toMatchSnapshot();
});

it('should get dates when interval is one hour', () => {
  const dateFrom = new Date('2018-01-10T00:00:00Z');
  const dateTo = new Date('2018-01-15T00:00:00Z');
  const dates = getDatesInRange(dateFrom, dateTo, ms('1 hour'));

  expect(dates).toMatchSnapshot();
});
