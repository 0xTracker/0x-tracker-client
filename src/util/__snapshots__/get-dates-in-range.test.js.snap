// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get dates when interval is one day 1`] = `
Array [
  2018-01-10T00:00:00.000Z,
  2018-01-11T00:00:00.000Z,
  2018-01-12T00:00:00.000Z,
  2018-01-13T00:00:00.000Z,
  2018-01-14T00:00:00.000Z,
  2018-01-15T00:00:00.000Z,
  2018-01-16T00:00:00.000Z,
  2018-01-17T00:00:00.000Z,
  2018-01-18T00:00:00.000Z,
  2018-01-19T00:00:00.000Z,
  2018-01-20T00:00:00.000Z,
  2018-01-21T00:00:00.000Z,
  2018-01-22T00:00:00.000Z,
  2018-01-23T00:00:00.000Z,
  2018-01-24T00:00:00.000Z,
  2018-01-25T00:00:00.000Z,
  2018-01-26T00:00:00.000Z,
  2018-01-27T00:00:00.000Z,
  2018-01-28T00:00:00.000Z,
  2018-01-29T00:00:00.000Z,
  2018-01-30T00:00:00.000Z,
  2018-01-31T00:00:00.000Z,
  2018-02-01T00:00:00.000Z,
  2018-02-02T00:00:00.000Z,
  2018-02-03T00:00:00.000Z,
  2018-02-04T00:00:00.000Z,
  2018-02-05T00:00:00.000Z,
  2018-02-06T00:00:00.000Z,
  2018-02-07T00:00:00.000Z,
  2018-02-08T00:00:00.000Z,
  2018-02-09T00:00:00.000Z,
  2018-02-10T00:00:00.000Z,
  2018-02-11T00:00:00.000Z,
  2018-02-12T00:00:00.000Z,
  2018-02-13T00:00:00.000Z,
  2018-02-14T00:00:00.000Z,
]
`;

exports[`should get dates when interval is one hour 1`] = `
Array [
  2018-01-10T00:00:00.000Z,
  2018-01-10T01:00:00.000Z,
  2018-01-10T02:00:00.000Z,
  2018-01-10T03:00:00.000Z,
  2018-01-10T04:00:00.000Z,
  2018-01-10T05:00:00.000Z,
  2018-01-10T06:00:00.000Z,
  2018-01-10T07:00:00.000Z,
  2018-01-10T08:00:00.000Z,
  2018-01-10T09:00:00.000Z,
  2018-01-10T10:00:00.000Z,
  2018-01-10T11:00:00.000Z,
  2018-01-10T12:00:00.000Z,
  2018-01-10T13:00:00.000Z,
  2018-01-10T14:00:00.000Z,
  2018-01-10T15:00:00.000Z,
  2018-01-10T16:00:00.000Z,
  2018-01-10T17:00:00.000Z,
  2018-01-10T18:00:00.000Z,
  2018-01-10T19:00:00.000Z,
  2018-01-10T20:00:00.000Z,
  2018-01-10T21:00:00.000Z,
  2018-01-10T22:00:00.000Z,
  2018-01-10T23:00:00.000Z,
  2018-01-11T00:00:00.000Z,
  2018-01-11T01:00:00.000Z,
  2018-01-11T02:00:00.000Z,
  2018-01-11T03:00:00.000Z,
  2018-01-11T04:00:00.000Z,
  2018-01-11T05:00:00.000Z,
  2018-01-11T06:00:00.000Z,
  2018-01-11T07:00:00.000Z,
  2018-01-11T08:00:00.000Z,
  2018-01-11T09:00:00.000Z,
  2018-01-11T10:00:00.000Z,
  2018-01-11T11:00:00.000Z,
  2018-01-11T12:00:00.000Z,
  2018-01-11T13:00:00.000Z,
  2018-01-11T14:00:00.000Z,
  2018-01-11T15:00:00.000Z,
  2018-01-11T16:00:00.000Z,
  2018-01-11T17:00:00.000Z,
  2018-01-11T18:00:00.000Z,
  2018-01-11T19:00:00.000Z,
  2018-01-11T20:00:00.000Z,
  2018-01-11T21:00:00.000Z,
  2018-01-11T22:00:00.000Z,
  2018-01-11T23:00:00.000Z,
  2018-01-12T00:00:00.000Z,
  2018-01-12T01:00:00.000Z,
  2018-01-12T02:00:00.000Z,
  2018-01-12T03:00:00.000Z,
  2018-01-12T04:00:00.000Z,
  2018-01-12T05:00:00.000Z,
  2018-01-12T06:00:00.000Z,
  2018-01-12T07:00:00.000Z,
  2018-01-12T08:00:00.000Z,
  2018-01-12T09:00:00.000Z,
  2018-01-12T10:00:00.000Z,
  2018-01-12T11:00:00.000Z,
  2018-01-12T12:00:00.000Z,
  2018-01-12T13:00:00.000Z,
  2018-01-12T14:00:00.000Z,
  2018-01-12T15:00:00.000Z,
  2018-01-12T16:00:00.000Z,
  2018-01-12T17:00:00.000Z,
  2018-01-12T18:00:00.000Z,
  2018-01-12T19:00:00.000Z,
  2018-01-12T20:00:00.000Z,
  2018-01-12T21:00:00.000Z,
  2018-01-12T22:00:00.000Z,
  2018-01-12T23:00:00.000Z,
  2018-01-13T00:00:00.000Z,
  2018-01-13T01:00:00.000Z,
  2018-01-13T02:00:00.000Z,
  2018-01-13T03:00:00.000Z,
  2018-01-13T04:00:00.000Z,
  2018-01-13T05:00:00.000Z,
  2018-01-13T06:00:00.000Z,
  2018-01-13T07:00:00.000Z,
  2018-01-13T08:00:00.000Z,
  2018-01-13T09:00:00.000Z,
  2018-01-13T10:00:00.000Z,
  2018-01-13T11:00:00.000Z,
  2018-01-13T12:00:00.000Z,
  2018-01-13T13:00:00.000Z,
  2018-01-13T14:00:00.000Z,
  2018-01-13T15:00:00.000Z,
  2018-01-13T16:00:00.000Z,
  2018-01-13T17:00:00.000Z,
  2018-01-13T18:00:00.000Z,
  2018-01-13T19:00:00.000Z,
  2018-01-13T20:00:00.000Z,
  2018-01-13T21:00:00.000Z,
  2018-01-13T22:00:00.000Z,
  2018-01-13T23:00:00.000Z,
  2018-01-14T00:00:00.000Z,
  2018-01-14T01:00:00.000Z,
  2018-01-14T02:00:00.000Z,
  2018-01-14T03:00:00.000Z,
  2018-01-14T04:00:00.000Z,
  2018-01-14T05:00:00.000Z,
  2018-01-14T06:00:00.000Z,
  2018-01-14T07:00:00.000Z,
  2018-01-14T08:00:00.000Z,
  2018-01-14T09:00:00.000Z,
  2018-01-14T10:00:00.000Z,
  2018-01-14T11:00:00.000Z,
  2018-01-14T12:00:00.000Z,
  2018-01-14T13:00:00.000Z,
  2018-01-14T14:00:00.000Z,
  2018-01-14T15:00:00.000Z,
  2018-01-14T16:00:00.000Z,
  2018-01-14T17:00:00.000Z,
  2018-01-14T18:00:00.000Z,
  2018-01-14T19:00:00.000Z,
  2018-01-14T20:00:00.000Z,
  2018-01-14T21:00:00.000Z,
  2018-01-14T22:00:00.000Z,
  2018-01-14T23:00:00.000Z,
]
`;

exports[`should get dates when interval is thirty minutes 1`] = `
Array [
  2018-01-10T00:00:00.000Z,
  2018-01-10T00:30:00.000Z,
  2018-01-10T01:00:00.000Z,
  2018-01-10T01:30:00.000Z,
  2018-01-10T02:00:00.000Z,
  2018-01-10T02:30:00.000Z,
  2018-01-10T03:00:00.000Z,
  2018-01-10T03:30:00.000Z,
  2018-01-10T04:00:00.000Z,
  2018-01-10T04:30:00.000Z,
  2018-01-10T05:00:00.000Z,
  2018-01-10T05:30:00.000Z,
  2018-01-10T06:00:00.000Z,
  2018-01-10T06:30:00.000Z,
  2018-01-10T07:00:00.000Z,
  2018-01-10T07:30:00.000Z,
  2018-01-10T08:00:00.000Z,
  2018-01-10T08:30:00.000Z,
  2018-01-10T09:00:00.000Z,
  2018-01-10T09:30:00.000Z,
  2018-01-10T10:00:00.000Z,
  2018-01-10T10:30:00.000Z,
  2018-01-10T11:00:00.000Z,
  2018-01-10T11:30:00.000Z,
  2018-01-10T12:00:00.000Z,
  2018-01-10T12:30:00.000Z,
  2018-01-10T13:00:00.000Z,
  2018-01-10T13:30:00.000Z,
  2018-01-10T14:00:00.000Z,
  2018-01-10T14:30:00.000Z,
  2018-01-10T15:00:00.000Z,
  2018-01-10T15:30:00.000Z,
  2018-01-10T16:00:00.000Z,
  2018-01-10T16:30:00.000Z,
  2018-01-10T17:00:00.000Z,
  2018-01-10T17:30:00.000Z,
  2018-01-10T18:00:00.000Z,
  2018-01-10T18:30:00.000Z,
  2018-01-10T19:00:00.000Z,
  2018-01-10T19:30:00.000Z,
  2018-01-10T20:00:00.000Z,
  2018-01-10T20:30:00.000Z,
  2018-01-10T21:00:00.000Z,
  2018-01-10T21:30:00.000Z,
  2018-01-10T22:00:00.000Z,
  2018-01-10T22:30:00.000Z,
  2018-01-10T23:00:00.000Z,
  2018-01-10T23:30:00.000Z,
  2018-01-11T00:00:00.000Z,
  2018-01-11T00:30:00.000Z,
  2018-01-11T01:00:00.000Z,
  2018-01-11T01:30:00.000Z,
  2018-01-11T02:00:00.000Z,
  2018-01-11T02:30:00.000Z,
  2018-01-11T03:00:00.000Z,
  2018-01-11T03:30:00.000Z,
  2018-01-11T04:00:00.000Z,
  2018-01-11T04:30:00.000Z,
  2018-01-11T05:00:00.000Z,
  2018-01-11T05:30:00.000Z,
  2018-01-11T06:00:00.000Z,
  2018-01-11T06:30:00.000Z,
  2018-01-11T07:00:00.000Z,
  2018-01-11T07:30:00.000Z,
  2018-01-11T08:00:00.000Z,
  2018-01-11T08:30:00.000Z,
  2018-01-11T09:00:00.000Z,
  2018-01-11T09:30:00.000Z,
  2018-01-11T10:00:00.000Z,
  2018-01-11T10:30:00.000Z,
  2018-01-11T11:00:00.000Z,
  2018-01-11T11:30:00.000Z,
  2018-01-11T12:00:00.000Z,
  2018-01-11T12:30:00.000Z,
  2018-01-11T13:00:00.000Z,
  2018-01-11T13:30:00.000Z,
  2018-01-11T14:00:00.000Z,
  2018-01-11T14:30:00.000Z,
  2018-01-11T15:00:00.000Z,
  2018-01-11T15:30:00.000Z,
  2018-01-11T16:00:00.000Z,
  2018-01-11T16:30:00.000Z,
  2018-01-11T17:00:00.000Z,
  2018-01-11T17:30:00.000Z,
  2018-01-11T18:00:00.000Z,
  2018-01-11T18:30:00.000Z,
  2018-01-11T19:00:00.000Z,
  2018-01-11T19:30:00.000Z,
  2018-01-11T20:00:00.000Z,
  2018-01-11T20:30:00.000Z,
  2018-01-11T21:00:00.000Z,
  2018-01-11T21:30:00.000Z,
  2018-01-11T22:00:00.000Z,
  2018-01-11T22:30:00.000Z,
  2018-01-11T23:00:00.000Z,
  2018-01-11T23:30:00.000Z,
  2018-01-12T00:00:00.000Z,
  2018-01-12T00:30:00.000Z,
  2018-01-12T01:00:00.000Z,
  2018-01-12T01:30:00.000Z,
  2018-01-12T02:00:00.000Z,
  2018-01-12T02:30:00.000Z,
  2018-01-12T03:00:00.000Z,
  2018-01-12T03:30:00.000Z,
  2018-01-12T04:00:00.000Z,
  2018-01-12T04:30:00.000Z,
  2018-01-12T05:00:00.000Z,
  2018-01-12T05:30:00.000Z,
  2018-01-12T06:00:00.000Z,
  2018-01-12T06:30:00.000Z,
  2018-01-12T07:00:00.000Z,
  2018-01-12T07:30:00.000Z,
  2018-01-12T08:00:00.000Z,
  2018-01-12T08:30:00.000Z,
  2018-01-12T09:00:00.000Z,
  2018-01-12T09:30:00.000Z,
  2018-01-12T10:00:00.000Z,
  2018-01-12T10:30:00.000Z,
  2018-01-12T11:00:00.000Z,
  2018-01-12T11:30:00.000Z,
  2018-01-12T12:00:00.000Z,
  2018-01-12T12:30:00.000Z,
  2018-01-12T13:00:00.000Z,
  2018-01-12T13:30:00.000Z,
  2018-01-12T14:00:00.000Z,
  2018-01-12T14:30:00.000Z,
  2018-01-12T15:00:00.000Z,
  2018-01-12T15:30:00.000Z,
  2018-01-12T16:00:00.000Z,
  2018-01-12T16:30:00.000Z,
  2018-01-12T17:00:00.000Z,
  2018-01-12T17:30:00.000Z,
  2018-01-12T18:00:00.000Z,
  2018-01-12T18:30:00.000Z,
  2018-01-12T19:00:00.000Z,
  2018-01-12T19:30:00.000Z,
  2018-01-12T20:00:00.000Z,
  2018-01-12T20:30:00.000Z,
  2018-01-12T21:00:00.000Z,
  2018-01-12T21:30:00.000Z,
  2018-01-12T22:00:00.000Z,
  2018-01-12T22:30:00.000Z,
  2018-01-12T23:00:00.000Z,
  2018-01-12T23:30:00.000Z,
  2018-01-13T00:00:00.000Z,
  2018-01-13T00:30:00.000Z,
  2018-01-13T01:00:00.000Z,
  2018-01-13T01:30:00.000Z,
  2018-01-13T02:00:00.000Z,
  2018-01-13T02:30:00.000Z,
  2018-01-13T03:00:00.000Z,
  2018-01-13T03:30:00.000Z,
  2018-01-13T04:00:00.000Z,
  2018-01-13T04:30:00.000Z,
  2018-01-13T05:00:00.000Z,
  2018-01-13T05:30:00.000Z,
  2018-01-13T06:00:00.000Z,
  2018-01-13T06:30:00.000Z,
  2018-01-13T07:00:00.000Z,
  2018-01-13T07:30:00.000Z,
  2018-01-13T08:00:00.000Z,
  2018-01-13T08:30:00.000Z,
  2018-01-13T09:00:00.000Z,
  2018-01-13T09:30:00.000Z,
  2018-01-13T10:00:00.000Z,
  2018-01-13T10:30:00.000Z,
  2018-01-13T11:00:00.000Z,
  2018-01-13T11:30:00.000Z,
  2018-01-13T12:00:00.000Z,
  2018-01-13T12:30:00.000Z,
  2018-01-13T13:00:00.000Z,
  2018-01-13T13:30:00.000Z,
  2018-01-13T14:00:00.000Z,
  2018-01-13T14:30:00.000Z,
  2018-01-13T15:00:00.000Z,
  2018-01-13T15:30:00.000Z,
  2018-01-13T16:00:00.000Z,
  2018-01-13T16:30:00.000Z,
  2018-01-13T17:00:00.000Z,
  2018-01-13T17:30:00.000Z,
  2018-01-13T18:00:00.000Z,
  2018-01-13T18:30:00.000Z,
  2018-01-13T19:00:00.000Z,
  2018-01-13T19:30:00.000Z,
  2018-01-13T20:00:00.000Z,
  2018-01-13T20:30:00.000Z,
  2018-01-13T21:00:00.000Z,
  2018-01-13T21:30:00.000Z,
  2018-01-13T22:00:00.000Z,
  2018-01-13T22:30:00.000Z,
  2018-01-13T23:00:00.000Z,
  2018-01-13T23:30:00.000Z,
  2018-01-14T00:00:00.000Z,
  2018-01-14T00:30:00.000Z,
  2018-01-14T01:00:00.000Z,
  2018-01-14T01:30:00.000Z,
  2018-01-14T02:00:00.000Z,
  2018-01-14T02:30:00.000Z,
  2018-01-14T03:00:00.000Z,
  2018-01-14T03:30:00.000Z,
  2018-01-14T04:00:00.000Z,
  2018-01-14T04:30:00.000Z,
  2018-01-14T05:00:00.000Z,
  2018-01-14T05:30:00.000Z,
  2018-01-14T06:00:00.000Z,
  2018-01-14T06:30:00.000Z,
  2018-01-14T07:00:00.000Z,
  2018-01-14T07:30:00.000Z,
  2018-01-14T08:00:00.000Z,
  2018-01-14T08:30:00.000Z,
  2018-01-14T09:00:00.000Z,
  2018-01-14T09:30:00.000Z,
  2018-01-14T10:00:00.000Z,
  2018-01-14T10:30:00.000Z,
  2018-01-14T11:00:00.000Z,
  2018-01-14T11:30:00.000Z,
  2018-01-14T12:00:00.000Z,
  2018-01-14T12:30:00.000Z,
  2018-01-14T13:00:00.000Z,
  2018-01-14T13:30:00.000Z,
  2018-01-14T14:00:00.000Z,
  2018-01-14T14:30:00.000Z,
  2018-01-14T15:00:00.000Z,
  2018-01-14T15:30:00.000Z,
  2018-01-14T16:00:00.000Z,
  2018-01-14T16:30:00.000Z,
  2018-01-14T17:00:00.000Z,
  2018-01-14T17:30:00.000Z,
  2018-01-14T18:00:00.000Z,
  2018-01-14T18:30:00.000Z,
  2018-01-14T19:00:00.000Z,
  2018-01-14T19:30:00.000Z,
  2018-01-14T20:00:00.000Z,
  2018-01-14T20:30:00.000Z,
  2018-01-14T21:00:00.000Z,
  2018-01-14T21:30:00.000Z,
  2018-01-14T22:00:00.000Z,
  2018-01-14T22:30:00.000Z,
  2018-01-14T23:00:00.000Z,
  2018-01-14T23:30:00.000Z,
]
`;
